### PR TITLE
added exchange and marketplace search UML

### PR DESCRIPTION
**Pull request by ghassanzadeh**
Added some classes for Exchange activities and Search in the MarketPlace that i thought is missing in our UML. I am not sure how much details i need to add to each class. Please let me know if i need to change any part and if the relationships make any sense! Thanks team :)

### CODE REVIEW

**Comment by ghassanzadeh** [comment]
Unlibrary_UML

**Review by calebschoepp — APPROVED** [review]
LGTM! Great work.

File: doc/p1-uml.txt [hunk 1]
```diff
@@ -80,4 +105,13 @@ GoogleMapsFragment o-- GoogleMapService
 ProfileFragment o-- ProfileFragmentViewModel
 ProfileFragmentViewModel --> LibraryViewModel
 ProfileFragment o-- BookRequestService
-@enduml
+
+FilterLibrary --> LibraryViewModel
+SearchMarketPlace --o FilterLibrary
+SearchMarketPlace o-- LibraryViewModel
+BorrowActivity --> ExchangeActivity
+BorrowActivity --o CameraService
```
Comment (cyrusdiego) [inline doc/p1-uml.txt:113]: I think the aggregation here is the opposite of what we'd want. The Camera Service would be a part of the BorrowActivity.

File: doc/p1-uml.txt [hunk 2]
```diff
@@ -80,4 +105,13 @@ GoogleMapsFragment o-- GoogleMapService
 ProfileFragment o-- ProfileFragmentViewModel
 ProfileFragmentViewModel --> LibraryViewModel
 ProfileFragment o-- BookRequestService
-@enduml
+
+FilterLibrary --> LibraryViewModel
+SearchMarketPlace --o FilterLibrary
+SearchMarketPlace o-- LibraryViewModel
```
Comment (cyrusdiego) [inline doc/p1-uml.txt:111]: I think for the MarketPlace Activity, we'd actually want to use a separate ViewModel and Repository, not the one from the Library

File: doc/p1-uml.txt [hunk 3]
```diff
@@ -1,4 +1,29 @@
 @startuml
+
+class ExchangeActivity{
+    #void onCreate()
+}
+
+class BorrowActivity{
+    -acceptRequest()
```
Comment (cyrusdiego) [inline doc/p1-uml.txt:8]: I think accepting / declining requests would only be under the library activity. I'm assuming this Borrow activity is when you click on a book to request to book, so then we'd only have a request method.

**Review by cyrusdiego — CHANGES_REQUESTED** [review]
Left some comments, I think for each path (library, unlibrary, marketplace, profile) we'd have separate viewmodels and repositories. So for the marketplace, we wouldn't want to use the libraries viewmodel as it would only house the books the owner has. For the marketplace we want a viewmodel and repository combo that holds all the books available to be borrowed.

File: doc/p1-uml.txt [hunk 4]
```diff
@@ -1,4 +1,29 @@
...
+
+class ReturnActivity{
+    -updateStatus()
+}
+
+class FilterLibrary{
+
+}
+
+class SearchMarketPlace{
```
Comment (armiantos) [inline doc/p1-uml.txt:22]: I would rename this to `MarketPlaceViewModel` and create a `MarketplaceActivity` that uses this so it's uniform with `LibraryActivity` and `ProfileActivity`

File: doc/p1-uml.txt (hunk 1 above)
Comment (armiantos) [reply doc/p1-uml.txt:113]: The borrow activity is a great idea!

**Review by armiantos — COMMENTED** [review]
Just nits to make the UML more readable. Great job!

File: doc/p1-uml.txt [hunk 5]
```diff
@@ -80,4 +93,9 @@ GoogleMapsFragment o-- GoogleMapService
 ProfileFragment o-- ProfileFragmentViewModel
 ProfileFragmentViewModel --> LibraryViewModel
 ProfileFragment o-- BookRequestService
+
+ExchangeActivity --o ExchangeViewModel
```
Comment (cyrusdiego) [inline doc/p1-uml.txt:97]: aggregation is backwards

**Review by cyrusdiego — CHANGES_REQUESTED** [review]

**Review by armiantos — APPROVED** [review]
Looks good to me!

**Review by cyrusdiego — APPROVED** [review]

**Pull request merged**